probe-rs/src/probe/stlink/mod.rs

```diff
@@ -487,7 +519,8 @@ impl STLink {
             &mut buf,
             TIMEOUT,
         )?;
-        Self::check_status(&buf)
+        Self::check_status(&buf)?;
+        Ok(())
```

Comment:
Why? Seems like `check_status` is always called at the end.

Reply:
The stlink returns a status code and here we check if it is okay or not. Basically poor mans ? operator ;)

Reply:
Yes, I mean, this looks bulkier than just that old `Self::check_status(&buf)`.

Reply:
Hmm then it cannot autoconverse the error type, no?

Reply:
Yep, for inference it is needed. Another way would be `.map(From::from)`, but then I like `?; Ok(())` more :)

Reply:
Ok, I was able to change it back to the old way =) Thanks!